did-doc-diff/src/index.ts

```diff
@@ -0,0 +1,217 @@
+import * as crypto from '@adxp/crypto'
+import { TID, util } from '@adxp/common'
+import { sha256 } from 'multiformats/hashes/sha2'
+import * as uint8arrays from 'uint8arrays'
+
+/*
+  Example identity tick
+    {
+      "tid": "3j55-hlo-v4g4-22",
+      "diffs": {
+        "3j55-hih-g24c-22": { "a": 1, "b": 2, "c": 3 },
+        "3j55-hiu-hcxs-22": {
+          "prev": "3j55-hih-g24c-22",
+          "patches": [["put", ["d"], 4]],
+          "key": "did:key:zDnaeYgdnK7nVH2xNQENrBRJbdCQ1KKjxt5sbykiUGLAh46Ez",
+          "sig": "z58kG77wksPLi8185y6fsJG3NidUjYVBzBKVqSG3T5NQKwrNu9d6aL2UorPRzre4VCvKUASuec53jain2LRaMCtmZ",
+        },
+        "3j55-hk2-pbqk-22": {
+          "prev": "3j55-hiu-hcxs-22",
+          "patches": [["del", ["b"]]],
+          "key": "did:key:zDnaeYgdnK7nVH2xNQENrBRJbdCQ1KKjxt5sbykiUGLAh46Ez",
+          "sig": "z45gdspAmr9W1tC2KCBrnugdQSXP7fbTUvxz8yM1QnPPNLTKHs2PWzDYgmJe8RPLhZKrfpVFEsvaCMHkhLkFGfxRo",
+        },
+        "3j55-hki-vs6w-22": {
+          "prev": "3j55-hk2-pbqk-22",
+          "patches": [["put", ["e", "ea"], "each"]],
+          "key": "did:key:zDnaeYgdnK7nVH2xNQENrBRJbdCQ1KKjxt5sbykiUGLAh46Ez",
+          "sig": "z66WkmrsuVNiqBT3mhjF8G5MNcgpcKbCvXYinqFL73xYkwbSRvNNE7Mj8ak9gEN5FCkamWgsa82GrG984sJxcJdw8",
+        },
+        "3j55-hl6-3ctt-22": {
+          "prev": "3j55-hk2-pbqk-22",
+          "patches": [["put", ["Don't", "put", "me"], "in"]],
+          "key": "did:key:zDnaeYgdnK7nVH2xNQENrBRJbdCQ1KKjxt5sbykiUGLAh46Ez",
+          "sig": "z5oHy2qCSKR9Z9hXuha6c7qEpUWMzGNeZcj8D5kDAxsWP6P8FN4kmSWHcoLv4q5umiyMWcUk5CePGeh51khRGEgGL",
+        }
+      },
+      "key": "did:key:zDnaesnjCLbxSB2EiJgTZUYCBqHTkuDJDdLbSmaAn2bpqXtuY",
+      "sig": "z4Y3LwMCQcmgFvyGfMmSCa5Qr7vCr7tUUDWnRBQqzZsDpSCKP3ZB1WnQTUVBivruavBsCprhjJ5p4WBR1RTyeYqUd"
+    }
+
+  This makes
+    {
+    id: 'did:aic:zwulnkrxrjkobowd',
+    a: 1,
+    c: 3,
+    d: 4,
+    e: { ea: 'each' },
+    }
+*/
+
+export async function tick_to_diffs(tick: {
```

Comment:
do camelCase instead of snake_case. for function names & variable names. so
`tickToDiffs`

Reply:
@AaronGoldman 
bump^^ we really do want to standardize on camelCase in code

Reply:
yup I will move them all to camelCase